zerver/views/push_notifications.py

```diff
@@ -101,3 +114,45 @@ def send_test_push_notification_api(
     send_test_push_notification(user_profile, devices)
 
     return json_success(request)
+
+
+@zulip_login_required
+def self_hosting_auth_redirect(request: HttpRequest) -> HttpResponse:
+    if not uses_notification_bouncer():
+        return render(request, "404.html", status=404)
+
+    user = request.user
+    assert user.is_authenticated
+    assert isinstance(user, UserProfile)
+    if not user.has_billing_access:
+        # We may want to replace this with an html error page at some point,
+        # but this endpoint shouldn't be accessible via the UI to an unauthorized
+        # user - and they need to directly enter the URL in their browser. So a json
+        # error may be sufficient.
+        raise OrganizationOwnerRequiredError
+
+    realm_info = get_realms_info_for_push_bouncer(user.realm_id)[0]
+
+    user_info = UserDataForRemoteBilling(
+        uuid=str(user.uuid),
+        email=user.delivery_email,
+        full_name=user.full_name,
+    )
+
+    post_data = {
+        "user": orjson.dumps(dict(user_info)).decode(),
```

Comment:
Pydantic models should be serialized directly with `user_info.model_dump_json()`, which avoids intermediate untyped `dict` steps.